### PR TITLE
MULE-13759: Fix CursorProvider resolution for ParameterGroups inline

### PR DESCRIPTION
**Pull request by alegmarra**

### CODE REVIEW

File: modules/extensions-support/src/main/java/org/mule/runtime/module/extension/internal/runtime/resolver/ResolverUtils.java [hunk 1]
```diff
@@ -87,6 +91,37 @@ private ResolverUtils() {
     return resolve;
   }
 
+  public static <T> T resolveValue(ValueResolver<T> resolver, ValueResolvingContext context)
+      throws MuleException {
+    T value = resolveRecursively(resolver, context);
+
+    if (value instanceof CursorProvider) {
```
Comment (marianogonzalez) [inline modules/extensions-support/src/main/java/org/mule/runtime/module/extension/internal/runtime/resolver/ResolverUtils.java]: at this point you should call resolveCursor, which is the one that should have the logic to handle the TypeValue

File: modules/extensions-support/src/main/java/org/mule/runtime/module/extension/internal/runtime/resolver/ResolverUtils.java [hunk 2]
```diff
@@ -87,6 +91,37 @@ private ResolverUtils() {
     return resolve;
   }
 
+  public static <T> T resolveValue(ValueResolver<T> resolver, ValueResolvingContext context)
```
Comment (marianogonzalez) [inline modules/extensions-support/src/main/java/org/mule/runtime/module/extension/internal/runtime/resolver/ResolverUtils.java]: jdoc

File: modules/extensions-support/src/main/java/org/mule/runtime/module/extension/internal/runtime/resolver/ResolverUtils.java [hunk 3]
```diff
@@ -87,6 +91,37 @@ private ResolverUtils() {
...
+            .type(cursor.getClass())
+            .mediaType(typedValue.getDataType().getMediaType())
+            .build(), typedValue.getLength());
+      }
+    }
+
+    return value;
+  }
+
+  public static Object resolveCursor(Object value) {
```
Comment (marianogonzalez) [inline modules/extensions-support/src/main/java/org/mule/runtime/module/extension/internal/runtime/resolver/ResolverUtils.java]: jdoc

File: modules/extensions-support/src/main/java/org/mule/runtime/module/extension/internal/runtime/resolver/ResolverUtils.java [hunk 4]
```diff
@@ -87,6 +91,37 @@ private ResolverUtils() {
     return resolve;
   }
 
+  public static <T> T resolveValue(ValueResolver<T> resolver, ValueResolvingContext context)
+      throws MuleException {
+    T value = resolveRecursively(resolver, context);
+
+    if (value instanceof CursorProvider) {
+      return (T) ((CursorProvider) value).openCursor();
+
```
Comment (marianogonzalez) [inline modules/extensions-support/src/main/java/org/mule/runtime/module/extension/internal/runtime/resolver/ResolverUtils.java]: empty lilne

Comment (alegmarra) [reply modules/extensions-support/src/main/java/org/mule/runtime/module/extension/internal/runtime/resolver/ResolverUtils.java]: intentional, reader friendly, emptyline... if the formatter allows it, it's legal

**Pull request merged**